### ISSUE TITLE
Namespace race condition

### ISSUE THREAD

**Issue by sepulworld** [lablabs/terraform-aws-eks-cluster-autoscaler#5]
Hi! Running into a namespace race condition

```
[2021-04-09T19:49:32Z] Error: create: failed to create: namespaces "cluster-autoscaler" not found
--
  | [2021-04-09T19:49:32Z]
  | [2021-04-09T19:49:32Z]   on .terraform/modules/eks.cluster_autoscaler/main.tf line 3, in resource "helm_release" "cluster_autoscaler":
  | [2021-04-09T19:49:32Z]    3: resource "helm_release" "cluster_autoscaler" {

```

Follow up run completes fine

**Comment by sepulworld** [comment]
Perhaps test out using https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#create_namespace instead of the k8s resource namespace creation.

**Comment by martinhaus** [comment]
Hi, thanks for pointing this out. Another solution would be to create a dependency between the k8s_namaspace resource and the helm release. That way we would ensure the namespace is finished creating before the helm release starts deploying.

Could you make a PR?

**Comment by dojci** [comment]
Closing this issues because it was Implemented in #7
